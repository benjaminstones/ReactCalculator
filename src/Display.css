.display-container {
        background-color: #606368;
}

h1 {
    font-family: 'Open Sans', sans-serif;
    margin: 0px;;
    padding-left: 3%;
    color: #F89124;
}