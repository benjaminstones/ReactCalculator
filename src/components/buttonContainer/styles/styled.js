import styled from 'styled-components';

export const StyledButtonRow = styled.div`
    height: 70px;
`
export const StyledButtonContainer = styled.div`
    > div {
        width: 100%;
        margin-bottom: 1px;
        flex: 1 0 auto;
        display: flex;
    }
`
