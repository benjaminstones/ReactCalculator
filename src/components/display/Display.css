.display-container {
        background-color: #606368;
}

h1 {
    font-family: 'Open Sans', sans-serif;
    font-size: 4rem;
    margin: 0px;;
    padding-left: 3%;
    color: #D0E6F4;
}