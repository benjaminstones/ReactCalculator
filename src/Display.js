import './Display.css'

function Display(props) {
  return (
    <div className='display-container'>
      <h1>{props.text}</h1>
    </div>
  );
}

export default Display;
