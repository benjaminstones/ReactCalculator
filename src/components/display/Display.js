import './Display.css';

const formatUnicode = (outputText) => {
	let formattedText = outputText.replaceAll('*', '\u00d7')
	formattedText = formattedText.replaceAll('/', '\u00f7')
	return(formattedText)
}

function Display(props) {
	return (
		<div className='display-container'>
			<h1>{formatUnicode(props.expression)}</h1>
			<h1>{props.total}</h1>
		</div>
	);
}

export default Display;
