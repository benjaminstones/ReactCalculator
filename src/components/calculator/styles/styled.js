import styled from 'styled-components';

export const StyledCalculatorContainer = styled.div`
    background-color: #202124;
    height: 100%;
    border: 5px solid black;
    width: 50%;
    margin: auto;
    margin-top: 50px;
    @media (max-width: 768px) {
        flex-direction: column;
        background-color: #202124;
        height: 100%;
        border: 5px solid black;
        width: 95%;
        height: 70%;
        margin: auto;
        margin-top: 50px;
    }
`