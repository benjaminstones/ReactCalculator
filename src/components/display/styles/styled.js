import styled from 'styled-components';

export const StyledDisplayContainer = styled.div`
    background-color: #606368;
`

export const StyledDisplayText = styled.h1`
    font-family: 'Open Sans', sans-serif;
    font-size: 4rem;
    margin: 0px;
    padding-left: 3%;
    color: #D0E6F4;
    width: 98%;
    word-break: break-word;
`