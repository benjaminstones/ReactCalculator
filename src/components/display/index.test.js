import React from 'react';
import { shallow } from 'enzyme';
import Display from './index.js'
import '../../setupTests'
import { StyledDisplayText } from './styles/styled.js';

let wrapped = shallow(<Display expression='2+3=' total='5' />);
describe('Display', () => {
	it('should render the Display Component correctly', () => {
		expect(wrapped).toMatchSnapshot();
	});
	it ('should render an expression and total in the display', () => {
		const expressionDisplay = wrapped.find('#expression-display');
		const totalDisplay = wrapped.find('#total-display')

		expect(wrapped.find(StyledDisplayText)).toHaveLength(2);
		expect(expressionDisplay.exists()).toBe(true);
		expect(totalDisplay.exists()).toBe(true);
	})
	it('should output the expression correctly', () => {
		const expressionDisplay = wrapped.find('#expression-display');

		expect(expressionDisplay.text()).toBe('2+3=')
	});
	it('should output the total correctly', () => {
		const totalDisplay = wrapped.find('#total-display')

		expect(totalDisplay.text()).toBe('5')
	});
});