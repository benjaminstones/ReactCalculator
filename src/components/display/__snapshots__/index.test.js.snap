// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display should output the Display Component correctly 1`] = `
<styled.div>
  <styled.h1
    id="expression-display"
  >
    2+3=
  </styled.h1>
  <styled.h1
    id="total-display"
  >
    5
  </styled.h1>
</styled.div>
`;

exports[`Display should output the expression correctly 1`] = `
<styled.div>
  <styled.h1
    id="expression-display"
  >
    2+3=
  </styled.h1>
  <styled.h1
    id="total-display"
  >
    5
  </styled.h1>
</styled.div>
`;

exports[`Display should render the Display Component correctly 1`] = `
<styled.div>
  <styled.h1
    id="expression-display"
  >
    2+3=
  </styled.h1>
  <styled.h1
    id="total-display"
  >
    5
  </styled.h1>
</styled.div>
`;

exports[`Display should render the Display Component correctly 2`] = `
<styled.div>
  <styled.h1
    id="expression-display"
  >
    2+3=
  </styled.h1>
  <styled.h1
    id="total-display"
  >
    5
  </styled.h1>
</styled.div>
`;
